### ISSUE TITLE
Windows Mill: control characters being outputted in PowerShell & Command Prompt - cannot read easily.

### ISSUE THREAD

**Issue by ScalaWilliam** [com-lihaoyi/mill#1284]
Hi, I am running Mill on Windows and this is what I get when running it:
![image](https://user-images.githubusercontent.com/2464813/115154534-38d37f00-a073-11eb-91dc-4fa088e48747.png)

Expectation: either display colors without control characters, or display no colors, so that the output is readable.

Running in Linux it works just fine (WSL).

Thanks!

**Comment by alexarchambault** [comment]
@ScalaWilliam Does it work when running mill via coursier? Like
```text
$ cs resolve mill:0.9.5 --scala 2.13.3 -- ..mill-args..
$ cs resolve mill-interactive:0.9.5 --scala 2.13.3 -- ..mill-args..
```

coursier enables ANSI output in the terminal, via [this library](https://github.com/alexarchambault/windows-ansi). If that fixes it, it could be interesting for mill to do the same.

**Comment by lefou** [comment]
@ScalaWilliam As a workaround, you can disable colored output with `--color false`.

**Comment by ScalaWilliam** [comment]
> @ScalaWilliam Does it work when running mill via coursier? Like
> 
> ```
> $ cs resolve mill:0.9.5 --scala 2.13.3 -- ..mill-args..
> $ cs resolve mill-interactive:0.9.5 --scala 2.13.3 -- ..mill-args..
> ```
> 
> coursier enables ANSI output in the terminal, via [this library](https://github.com/alexarchambault/windows-ansi). If that fixes it, it could be interesting for mill to do the same.

That didn't work - did you mean the below for example?
```
> cs.exe launch mill-interactive:0.9.5 -- bsp.test
```

In which case the colour bug was not there.
However, it would hang (I tried 0.9.6 as well) on this stage: `[5/59] mill.scalalib.ZincWorkerModule.classpath`. I tried it both on `mill` and on my own mini test project.



> @ScalaWilliam As a workaround, you can disable colored output with `--color false`.

@lefou this seems to not have worked.
```
> mill --color false foo.test
[4/59] foo.test.resources
←[1A←[2K←[9999D[8/59] foo.sources
←[1A←[2K←[9999D[13/59] foo.resources
←[1A←[2K←[9999D[32/59] foo.test.sources
...
```

**Comment by lefou** [comment]
> @lefou this seems to not have worked.
> 
> ```
> > mill --color false foo.test
> [4/59] foo.test.resources
> ←[1A←[2K←[9999D[8/59] foo.sources
> ←[1A←[2K←[9999D[13/59] foo.resources
> ←[1A←[2K←[9999D[32/59] foo.test.sources
> ...
> ```

Looks like these are the control sequences used to overwrite the same line with the updated progress, but not the colors.

**Comment by alexarchambault** [comment]
> In which case the colour/special character bug was not there.

That means [windows-ansi](https://github.com/alexarchambault/windows-ansi) worked.

@lefou [This call](https://github.com/alexarchambault/windows-ansi/tree/c3fbbb59def1a55903853717b972801b0445c92e#change-terminal-mode) of a windows-ansi method allows to print ANSI escape codes in the terminal under Windows. (This calls a system call via JNI, using a DLL from jline, see the windows-ansi README.)

**Comment by ScalaWilliam** [comment]
Many people in my circles use Windows and we'd be one step away from getting more Mill users if this got fixed :-)

**Comment by lefou** [comment]
@ScalaWilliam I just merged #1308 which should fix the issue. Please give it a go (after it got published) and report back, if you still have issues.
